pkg/controller/shared_state.go

```diff
@@ -86,8 +88,30 @@ type SharedState struct {
 	azureDiskCSITranslator        csitranslator.InTreePlugin
 	availableAttachmentsMap       sync.Map
 	driverLifecycle               DriverLifecycle
+	eventsToPersistQueue          chan ReplicaAttachmentFailureInfo
+	eventsToUnpersistQueue        chan string
+	_eventRefresherInitOnce       sync.Once
 }
 
+type ReplicaAttachmentFailureInfo struct {
+	volumeName string
+	message    string
+	pods       []runtime.Object
+	timestamp  time.Time
+}
+
+const (
+	// persistent event and their refreshes should overlap by this amount of time
+	eventOverlapDuration = time.Millisecond
```

Comment:
We should probably use a larger overlap time. During periods of heavy server activity, throttling could result in delays over 1s. 30s would probably be a reasonable value to ensure the event is always visible, but minimize the period in which a duplicate may appear.